### PR TITLE
devx: document work leads process

### PR DESCRIPTION
**Pull request by bobheadxi**
Documents the [Work lead proposal](https://docs.google.com/document/d/17gtVR_iS-RNko525s7Ib2fGpBSomBnc3i-uDAP00gI4/edit#bookmark=id.vp63ni32islq) we've been discussing as framework for planning.

Preview: https://deploy-preview-2115--sourcegraph-handbook.netlify.app/departments/product-engineering/engineering/enablement/dev-experience#planning

### CODE REVIEW

File: content/departments/product-engineering/engineering/enablement/dev-experience/index.md [hunk 1]
```diff
@@ -71,13 +71,60 @@ To learn more about contributing to `sg`, check out the [contribution guide](htt
...
+
+A **work lead** is the person assigned to lead a body of work, including the planning of said work.
+If there are questions about priority order, scope change, etc. the work lead helps answer questions/provide updates.
+They have special responsibilities regarding [planning issues](#planning-issues) and [tracking issues](#tracking-issues).
+
+#### Planning issues
+
+Planning starts with a **planning issue**. Planning issues are GitHub issues with the [`team/devx` and `planning` labels](https://github.com/sourcegraph/sourcegraph/issues?q=sort%3Aupdated-desc+is%3Aissue+label%3Aplanning+label%3Ateam%2Fdevx) on our [GitHub board](https://github.com/orgs/sourcegraph/projects/212) to track any significant planning efforts. A planning issue documents the planning process - links to RFC(s), discussions, artefacts like scratch documents, [meetings](#meetings), decisions, etc. as planning progresses.
+
+> NOTE: An updated and well-documented planning issue makes it easier to teammates and stakeholders to pick up context around the state of planning, and helps represent the often significant time investment involved with planning efforts.
```
Comment (taylorsperry) [inline content/departments/product-engineering/engineering/enablement/dev-experience/index.md:93]: This is great. I wonder if it's worth saying something explicit like, "A stakeholder should be able to look at a planning issue and get a high-level understanding of the problem we're trying to solve and the steps we're taking to address it, without having to read through the linked resources." I want to make sure we avoid having the planning issue just be a list of aggregated links.

File: content/departments/product-engineering/engineering/enablement/dev-experience/index.md [hunk 2]
```diff
@@ -71,13 +71,60 @@ To learn more about contributing to `sg`, check out the [contribution guide](htt
...
+- Help/work with PM/EM to:
+  - Define **Key Results**, including:
+    - Desired outcomes of this work
+    - Estimation that indicates the work a manageable size for one team to deliver within one quarter (timeboxed)
+    - How we will accomplish "Non-Functional Requirements" where applicable (or call out as N/A): testing, monitoring (how do we monitor and support the system?), documentation, etc.
+  - Identify:
+    - Stakeholders to notify and consult
+    - Dependencies on other teams and/or other work
+
+**✅ Outcome**: Once a roadmap or scope has been finalized, the appropriate tasks should be created, labelled with days estimates ([as supported by the tracking-issue-bot](../../process/tracking_issues.md#labels)) and other metadata where helpful, as well as a rough order of operation and outline of what can be parallelized. A [tracking issue](../../process/tracking_issues.md) should then be created.
```
Comment (taylorsperry) [inline content/departments/product-engineering/engineering/enablement/dev-experience/index.md:107]: Boom, I love this.

File: content/departments/product-engineering/engineering/enablement/dev-experience/index.md [hunk 3]
```diff
@@ -71,13 +71,60 @@ To learn more about contributing to `sg`, check out the [contribution guide](htt
...
+
+- Continuously document the planning process
+- Help/work with PM/EM to:
+  - Define **Key Results**, including:
+    - Desired outcomes of this work
+    - Estimation that indicates the work a manageable size for one team to deliver within one quarter (timeboxed)
+    - How we will accomplish "Non-Functional Requirements" where applicable (or call out as N/A): testing, monitoring (how do we monitor and support the system?), documentation, etc.
+  - Identify:
+    - Stakeholders to notify and consult
+    - Dependencies on other teams and/or other work
```
Comment (jhchabran) [inline content/departments/product-engineering/engineering/enablement/dev-experience/index.md]: A common problem on this specific point is for a work lead to exhibit the said dependency on another team but when the workload increases past a certain threshold, to passively slip into not taking action toward either polling its status or making sure it gets unblocked. 

```suggestion
  - Identify and monitor:
    - Stakeholders to notify and consult
    - Dependencies on other teams and/or other work and their progress status. 
```

File: content/departments/product-engineering/engineering/enablement/dev-experience/index.md (hunk 1 above)
Comment (jhchabran) [reply content/departments/product-engineering/engineering/enablement/dev-experience/index.md:93]: I agree with @taylorsperry here, I like the suggested changes because they clarify what a good description is. Ideally, by describing it as an ELI5 (for the lack of a better term), it makes it easy for everyone involved to discuss the task without having to come up with an abstract on the fly. And if we can't ELI5 it, that's a smell!

File: content/departments/product-engineering/engineering/enablement/dev-experience/index.md (hunk 2 above)
Comment (jhchabran) [reply content/departments/product-engineering/engineering/enablement/dev-experience/index.md:107]: Just a minor nitpick, I've often observed that days estimate are carrying the wrong meaning because they conflate a timeline on top of how much work there is. Looking at an issue with let's say about one day of work may induce some judgment errors on the timing when the assignee is handling a lot of support work for example. 

Just to be clear, I think we're perfectly fine with days to start, there's no need to overthink the process at this stage, we can always refine it later 😊

**Review by jhchabran — APPROVED** [review]
LGTM! I commented on a few points, but nothing that prevents moving forward :)

File: content/departments/product-engineering/engineering/enablement/dev-experience/index.md (hunk 1 above)
Comment (bobheadxi) [reply content/departments/product-engineering/engineering/enablement/dev-experience/index.md:93]: > A stakeholder should be able to look at a planning issue and get a high-level understanding of the problem we're trying to solve and the steps we're taking to address it, without having to read through the linked resources

I think this is something better for the Tracking issue perhaps? That is when the problem space and the steps we are taking to address it become more finalized.

Alternatively, I could add the following:

- Planning issues should clearly state the problem space and what we are hoping to achieve.
- Tracking issues should indicate the high-level steps we've decided to take to tackle the problems outlined in planning.

File: content/departments/product-engineering/engineering/enablement/dev-experience/index.md (hunk 2 above)
Comment (bobheadxi) [reply content/departments/product-engineering/engineering/enablement/dev-experience/index.md:107]: I think the days issue is okay! I find days of work an adequate abstraction - it indicates days of work on the issue itself, which should exclude support work etc. because that's not spent on the issue itself

File: content/departments/product-engineering/engineering/enablement/dev-experience/index.md [hunk 4]
```diff
@@ -71,13 +71,60 @@ To learn more about contributing to `sg`, check out the [contribution guide](htt
...
+
+> NOTE: An updated and well-documented planning issue makes it easier to teammates and stakeholders to pick up context around the state of planning, and helps represent the often significant time investment involved with planning efforts.
+
+The [work lead](#work-leads) should:
+
+- Continuously document the planning process
+- Help/work with PM/EM to:
+  - Define **Key Results**, including:
+    - Desired outcomes of this work
+    - Estimation that indicates the work a manageable size for one team to deliver within one quarter (timeboxed)
```
Comment (marekweb) [inline content/departments/product-engineering/engineering/enablement/dev-experience/index.md]:
```suggestion
    - Scoping the work to a manageable size for one team to deliver within one quarter (timeboxed)
```
